Регистрируемся на сайте

M2rk13 opened the pull request:
---
:mortar_board: [Регистрируемся на сайте](https://up.htmlacademy.ru/php/12/user/1415187/tasks/14)

M2rk13 committed:
Почти закончено 7.1

M2rk13 committed:
Закончено задание 7.1

M2rk13 committed:
Значительные улучшения кода

M2rk13 committed:
-

M2rk13 committed:
Merge branch 'master' into module7-task1

M2rk13 committed:
Внесены исправления для валидации фото

M2rk13 committed:
Улучшено оформление кода

functions.php приведен в общий вид
логика подключения к БД перенесена в функцию
исправлена ошибка отсутствия валидации формы видео

M2rk13 committed:
Merge branch 'master' into module7-task1

M2rk13 committed:
Восстановлена утерянная функция

M2rk13 committed:
-

00dev commented on functions.php:
```diff
@@ -83,60 +124,458 @@ function filter_size_ico($type) {
+
+    if ($con == false) {
+        $error = mysqli_connect_error();
+        print($error);
```
Вот так делать очень не рекомендую. Ошибки вообще напрямую лучше не показывать пользователю. Особенно ошибки подключения к базе. Движки баз данных часто показывают параметры подключения в тексте ошибки.  Там могут быть имя и пароль подключения

M2rk13 committed:
Мелкие исправления

Исправлен вывод ошибки подключения к БД,
Функция sequre_query_bind_result оптимизирована для замены sequre_query
Исправлена опечатка в правилах валидации для добавления поста цитаты
Применен патч, исправляющий загрузку файла.
